MaxText/layers.py

```diff
@@ -113,6 +117,43 @@ def dot_product_attention(query: Array,
   assert key.shape[-3] == value.shape[-3], 'k, v lengths must match.'
   assert query.shape[-1] == key.shape[-1], 'q, k depths must match.'
 
+  def compute_qk_attn_weights(query, key, cfg, aqt_rng):
```

Comment:
These 2 functions are almost identical, factor out common code?
You can put cfg.fwd_int8_qk and alike into params.

Reply:
I actually don't mind this duplication. This hides complexity fully from the reader and they might diverge more in the future. Unless you feel strongly Lukasz, let's say this is ok.